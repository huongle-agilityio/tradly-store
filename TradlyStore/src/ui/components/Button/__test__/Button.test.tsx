import {render, fireEvent, screen} from '@testing-library/react-native';
import {Text} from 'react-native';

//  Components
import {Button} from '..';

describe('Button Component', () => {
  it('Should Should renders correctly with children', () => {
    const container = render(<Button>Click me</Button>).toJSON();

    expect(screen.getByText('Click me')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('Should Should renders correctly when variant is not solid', () => {
    const container = render(
      <Button variant="bordered">Click me</Button>,
    ).toJSON();

    expect(screen.getByText('Click me')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('Should Should renders correctly when variant is solid and color is secondary', () => {
    const container = render(
      <Button color="secondary">Click me</Button>,
    ).toJSON();

    expect(screen.getByText('Click me')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('Should Should renders correctly when variant is not solid and color is secondary', () => {
    const container = render(
      <Button color="secondary" variant="bordered">
        Click me
      </Button>,
    ).toJSON();

    expect(screen.getByText('Click me')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('Should calls onPress when pressed', () => {
    const mockOnPress = jest.fn();
    render(<Button onPress={mockOnPress}>Click</Button>);

    fireEvent.press(screen.getByTestId('button'));

    expect(mockOnPress).toHaveBeenCalled();
  });

  it('Should does not call onPress when disabled', () => {
    const mockOnPress = jest.fn();
    render(
      <Button onPress={mockOnPress} disabled>
        Click
      </Button>,
    );

    fireEvent.press(screen.getByTestId('button'));

    expect(mockOnPress).not.toHaveBeenCalled();
  });

  it('Should does not call onPress when loading', () => {
    const mockOnPress = jest.fn();
    render(
      <Button onPress={mockOnPress} isLoading>
        Click
      </Button>,
    );

    fireEvent.press(screen.getByTestId('button'));
    expect(mockOnPress).not.toHaveBeenCalled();
  });

  it('Should displays Loading text when isLoading is true', () => {
    render(<Button isLoading>Click</Button>);

    expect(screen.getByTestId('button-loading')).toBeTruthy();
  });

  it('Should renders icon if provided', () => {
    render(<Button icon={<Text testID="icon">Icon</Text>}>Click</Button>);

    expect(screen.getByTestId('icon')).toBeTruthy();
  });
});
