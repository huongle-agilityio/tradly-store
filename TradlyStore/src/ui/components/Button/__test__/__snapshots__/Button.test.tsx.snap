// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should Should renders correctly when variant is not solid 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderColor": "#33907c",
      "borderRadius": 24,
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 23,
      "paddingVertical": 4,
      "position": "relative",
    }
  }
  testID="button"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat",
          },
          {},
          {
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 12,
            "fontWeight": "700",
            "lineHeight": 21,
          },
          {
            "color": "#33907c",
          },
          undefined,
        ]
      }
    >
      Click me
    </Text>
  </View>
</View>
`;

exports[`Button Component Should Should renders correctly when variant is not solid and color is secondary 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderColor": "#fff",
      "borderRadius": 24,
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 23,
      "paddingVertical": 4,
      "position": "relative",
    }
  }
  testID="button"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat",
          },
          {},
          {
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 12,
            "fontWeight": "700",
            "lineHeight": 21,
          },
          {
            "color": "#fff",
          },
          undefined,
        ]
      }
    >
      Click me
    </Text>
  </View>
</View>
`;

exports[`Button Component Should Should renders correctly when variant is solid and color is secondary 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#fff",
      "borderRadius": 24,
      "borderWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 23,
      "paddingVertical": 4,
      "position": "relative",
    }
  }
  testID="button"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat",
          },
          {
            "color": "#fff",
          },
          {
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 12,
            "fontWeight": "700",
            "lineHeight": 21,
          },
          {
            "color": "#33907c",
          },
          undefined,
        ]
      }
    >
      Click me
    </Text>
  </View>
</View>
`;

exports[`Button Component Should Should renders correctly with children 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#33907c",
      "borderColor": "#33907c",
      "borderRadius": 24,
      "borderWidth": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 23,
      "paddingVertical": 4,
      "position": "relative",
    }
  }
  testID="button"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat",
          },
          {
            "color": "#fff",
          },
          {
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 12,
            "fontWeight": "700",
            "lineHeight": 21,
          },
          {
            "color": "#fff",
          },
          undefined,
        ]
      }
    >
      Click me
    </Text>
  </View>
</View>
`;
